### PR TITLE
GH201,GH202: Template uri, blob & markdown

### LINKED ISSUE #201: Make blob filename configurable

### LINKED ISSUE #202: Make markdown tag/image url configurable

### PR DESCRIPTION
**Pull request by devlead**
* Adds template support for
    * Markdown
    * Blob name/path
    * Blob uri
* README
  * Adds config section
  * Fixes som MD linting issues
* fixes #201
* fixes #202

### CODE REVIEW

**Comment by devlead** [comment]
@gep13 rebased.

File: README.md [hunk 1]
```diff
@@ -38,33 +38,77 @@ choco install clipimg-vscode
...
+
+Templates provide a powerful way to customize the markdown, names, uris, etc. generated to suit your needs.
+The liquid template language is what's used and you can read more about it at LiquidJS [
+Introduction to Liquid Template Language](https://liquidjs.com/tutorials/intro-to-liquid.html).
+
+| Key                               | Default value                                     | Description                          |
+|-----------------------------------|---------------------------------------------------|--------------------------------------|
+| `clipImg.templateMarkdown`        | `![{{alt}}]({{uri}})`                             | Controls the markdown inserted       |
+| `clipImg.templateBlobName`        | `{{\"now\" | date: \"%Y/%m/%d\"}}/{{filename}}`   | Controls the naming of the blob path |
+| `clipImg.templateBlobUri`         | `{{uri}}?{{sas}}`                                 | Controls the uri used within markdown tag, lets you i.e. exclude sas token and change base uri, which ise useful i.e. when using a CDN in-front of the storage account.  |
```
Comment (gep13) [inline README.md:89]:
```suggestion
| `clipImg.templateBlobUri`         | `{{uri}}?{{sas}}`                                 | Controls the uri used within markdown tag, lets you i.e. exclude sas token and change base uri, which is useful i.e. when using a CDN in-front of the storage account.  |
```

File: README.md [hunk 2]
```diff
@@ -38,33 +38,77 @@ choco install clipimg-vscode
...
+
+#### Example
+
+In the below example the alt text the first letter of alt test will always be upper case i.e. `car` will become `Car` when inserted, filename blob file name will use `Year/Month/Day/HourMinuteSecond_Filename` (i.e. `2020/09/08/085110_75f8e3ed-5340-872c-3f18-69c985e76311.png`) as blob name, and cdn for the uri used (i.e. `https://mycdnurl.azureedge.net/public/2020/09/08/085110_75f8e3ed-5340-872c-3f18-69c985e76311.png`).
```
Comment (gep13) [inline README.md:93]:
```suggestion
In the below example the first letter of alt text will always be upper case i.e. `car` will become `Car` when inserted, filename blob file name will use `Year/Month/Day/HourMinuteSecond_Filename` (i.e. `2020/09/08/085110_75f8e3ed-5340-872c-3f18-69c985e76311.png`) as blob name, and cdn for the uri used (i.e. `https://mycdnurl.azureedge.net/public/2020/09/08/085110_75f8e3ed-5340-872c-3f18-69c985e76311.png`).
```

**Review by gep13 — CHANGES_REQUESTED** [review]

**Comment by devlead** [comment]
@gep13 addressed feedback and rebased.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@devlead thanks for getting this done!

**Pull request merged**